# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/client/TextureRecorded.h
# Commit: de6ddf5eb2a7
# Full Hash: de6ddf5eb2a7bd3d8a2f58636760e0aa16c584b9
# Author: Lee Salzman <lsalzman@mozilla.com>
# Date: 2023-12-05 21:29:44
# Regressor Bug: 1829026
# File Overlap Count: 1
# Description:
#   Bug 1829026 - Update CanvasTranslator to work with DrawTargetWebgl. r=aosmond
#   
#   This adds the necessary infrastructure for CanvasTranslator to allocate DrawTargetWebgl
#   instead of just allocating TextureData, and to use RemoteTextureMap to handle sending
#   the DrawTargetWebgl frames to the compositor.
# ==============================================================================

diff -r 773e7b31ac6e -r de6ddf5eb2a7 gfx/layers/client/TextureRecorded.h
--- a/gfx/layers/client/TextureRecorded.h	Tue Dec 05 09:37:03 2023 +0000
+++ b/gfx/layers/client/TextureRecorded.h	Tue Dec 05 09:37:03 2023 +0000
@@ -40,6 +40,11 @@
 
   TextureFlags GetTextureFlags() const final;
 
+  void SetRemoteTextureOwnerId(
+      RemoteTextureOwnerId aRemoteTextureOwnerId) final;
+
+  bool RequiresRefresh() const final;
+
  private:
   DISALLOW_COPY_AND_ASSIGN(RecordedTextureData);
 
@@ -51,7 +56,10 @@
   gfx::SurfaceFormat mFormat;
   RefPtr<gfx::DrawTarget> mDT;
   RefPtr<gfx::SourceSurface> mSnapshot;
+  ThreadSafeWeakPtr<gfx::SourceSurface> mSnapshotWrapper;
   OpenMode mLockedMode;
+  layers::RemoteTextureId mLastRemoteTextureId;
+  layers::RemoteTextureOwnerId mRemoteTextureOwnerId;
 };
 
 }  // namespace layers